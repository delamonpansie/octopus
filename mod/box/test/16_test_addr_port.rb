#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

class Env < RunEnv
  def config
    super.gsub(/.*_port.*/, '') + "\n" + $x + "\n"
  end
end


$x = "primary_port = 33013"
Env.connect_eval do ping end

$x = "primary_addr = 33013"
Env.new.env_eval do
  invoke :setup
  puts `./octopus 2>&1`.match(/Option 'primary_port' is not set/)[0]
  puts
end

$x = 'primary_addr = "127.0.0.1"'
Env.new.env_eval do
  invoke :setup
  puts `./octopus 2>&1`.match(/Option 'primary_port' is not set/)
  puts
end

$x = "primary_addr = \"127.0.0.1:33013\"\nprimary_port = 33013"
Env.connect_eval do ping end

$x = "primary_addr = \"127.0.0.1\"\nprimary_port = 33013"
Env.connect_eval do ping end

$x = 'primary_addr = "127.0.0.1:33013"'
Env.connect_eval do ping end

$x = "primary_addr = \"127.0.0.1:13\"\nprimary_port = 33013"
Env.connect_eval do
  ping
  puts File.new('octopus.log').read.match(/Option 'primary_addr' is overridden by 'primary_port'/)
end
