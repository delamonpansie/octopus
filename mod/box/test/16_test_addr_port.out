# box.ping()
:pong

Option 'primary_port' is not set

Option 'primary_port' is not set

# box.ping()
:pong

# box.ping()
:pong

# box.ping()
:pong

# box.ping()
:pong

Option 'primary_addr' is overridden by 'primary_port'
